### PR TITLE
[#ENTE-74] new readonlyNonEmptySetType

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes
new `readonlyNonEmptySetType` defined
Unit test for `readonlyNonEmptySetType` decoder
#### Motivation and Context
Costom decoder for `readonlyNonEmptySetType` usable into `Service` model to verify non-empty `authorizedCIDRs` ([link](https://github.com/pagopa/io-functions-commons/blob/master/src/models/service.ts#L76)).

#### How Has This Been Tested?
unit test

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/__tests__/types.test.ts [hunk 1]
```diff
@@ -39,6 +39,33 @@ describe("readonlySetType", () => {
   });
 });
 
+describe("readonlyNonEmptySetType", () => {
+  const aReadonlyNonEmptySet = readonlyNonEmptySetType(t.string, "Set of strings");
+
+  it("should not validate if is an empty set", () => {
```
Comment (balanza) [inline src/__tests__/types.test.ts:45]: Non blocking: we can use `it.each`:
```ts
it.each`
  title | f
  ${"an empty set"} | ${new Set()}
  ${"an empty array"} | ${[]}
`("should not validate if is $title", ({ f }) => {
      const v = aReadonlyNonEmptySet.decode(f);
      expect(v.isLeft()).toBeTruthy();
      expect(aReadonlyNonEmptySet.is(f)).toBeFalsy();
  });
```

**Review by AleDore — COMMENTED** [review]
It seems tests fail. BTW the type's implementation LGTM :)

File: src/__tests__/types.test.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/__tests__/types.test.ts]: fixed in 77d9cb7

File: src/types.ts [hunk 2]
```diff
@@ -84,10 +84,38 @@ export const readonlySetType = <E>(
...
+
+/**
+ * Creates an io-ts Type from a ReadonlyNonEmptySet
+ */
+export const readonlyNonEmptySetType = <E>(
+  o: t.Type<E, t.mixed>,
+  name: string
+): t.Type<ReadOnlyNonEmptySet<E>, t.mixed> => {
+  const arrayType = t.readonlyArray(o, name);
+  return new t.Type<ReadOnlyNonEmptySet<E>, t.mixed>(
```
Comment (balanza) [inline src/types.ts:110]: what about:

```ts
t.refinement(readonlySetType(o, name), e => e.size > 0);
```

Comment (BurnedMarshal) [reply src/types.ts]: I like it, fixed in 5fdc814

**Comment by BurnedMarshal** [comment]
@balanza @AleDore can we merge this PR?

**Review by AleDore — APPROVED** [review]
LGTM

**Review by balanza — APPROVED** [review]

**Pull request merged**